docs/3.0rc/develop/write-tasks/index.mdx

```diff
@@ -483,148 +483,223 @@ def my_flow():
 
 ## Caching
 
-Tasks support the ability to cache their return value.
-Caching allows you to efficiently reuse [results](/3.0rc/develop/interact-with-data/) of tasks that may be 
-expensive to reproduce with every flow run. You can also reuse cached results if the inputs to a task have 
-not changed.
+Caching refers to the ability of a task run to enter a `Completed` state and return a predetermined 
+value without actually running the code that defines the task. 
+This allows you to efficiently reuse results of tasks that may be expensive to compute
+and ensure that your pipelines are idempotent when retrying them due to unexpected failure. 
 
-Define a task that is cached based on its inputs by using the Prefect 
-[`task_input_hash`](/3.0rc/api-ref/prefect/tasks/#prefect.tasks.task_input_hash).
+By default Prefect's caching logic is based on the following attributes of a task invocation:
+- the inputs provided to the task
+- the code definition of the task
+- the prevailing flow run ID, or if executed autonomously, the prevailing task run ID
+
+All of these are hashed to compute the task's _cache key_.
```

Comment:
I think it is important to acknowledge the expected consequence is that a task called twice in the same flow with the same arguments will not run twice but reuse its cached results -- inter-flow idempotency.

Reply:
Good call, will do

Reply:
@jlowin updated